### ISSUE TITLE
Download folder file name

### ISSUE THREAD

**Issue by rms95** [microsoft/vscode-remote-release#7944]
Type: <b>Bug</b>

I have a folder on my remote Visual Studio Code I wish to download to my local machine. I do this by right clicking the folder in the "Explorer" view and selecting "Download...". The save window takes over the name of the selected folder, though it appears it adds some kind of invalid character to the end of the filename. If you then click save, it says "The file name is not valid". If I delete the added character, it saves successfully.

Extension version: 0.23.0
VS Code version: Code 1.74.3 (97dec172d3256f8ca4bfb2143f3f76b503ca0534, 2023-01-09T16:59:02.252Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No
Remote OS version: Linux x64 5.10.0-21-amd64

<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
vscccc:30610679
pyindex848cf:30577861
nodejswelcome1cf:30587006
2e4cg342:30602488
89544117:30613380
pythonsymbol12cf:30652349

```

</details>

**Comment by rms95** [comment]
Some additional info:
It appears the filename on Linux side contained characters which aren't valid for Windows. Therefore this probably isn't that much of an issue; though normalizing the file to the target OS would reduce confusion.

**Comment by lramos15** [comment]
How could we normalize the file? Do you recommend just stripping out the invalid characters? That may fundamentally change the file and it seems better to allow the user to know it is invalid and allow them to change the name as they see fit.

**Comment by rms95** [comment]
You are right about that. I later figured out the underlying issue in my case, where a carriage return appeared at the end of the file name (because the script making the file had windows line endings), which is a valid filename char in Linux, but not in windows. This kind of issue can indeed not really be normalized.
